engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -154,29 +174,81 @@ private boolean typeCanOverride(MethodRootNode node, EnsoContext ctx) {
     Type warning = builtins.warning();
     Type panic = builtins.panic();
     return methodOwnerType.isEigenType()
-            && builtins.nothing() != methodOwnerType
-            && any.getEigentype() != methodOwnerType
-            && panic.getEigentype() != methodOwnerType
-            && warning.getEigentype() != methodOwnerType;
+        && builtins.nothing() != methodOwnerType
+        && any.getEigentype() != methodOwnerType
+        && panic.getEigentype() != methodOwnerType
+        && warning.getEigentype() != methodOwnerType;
   }
 
   @Specialization(
-          replaces = "doFunctionalDispatchCachedSymbol",
-          guards = {"typesLibrary.hasType(self)", "!typesLibrary.hasSpecialDispatch(self)"})
+      replaces = "doFunctionalDispatchCachedSymbol",
+      guards = {"typesLibrary.hasType(self)", "!typesLibrary.hasSpecialDispatch(self)"})
   Object doFunctionalDispatchUncachedSymbol(
-          VirtualFrame frame,
-          State state,
-          UnresolvedSymbol symbol,
-          Object self,
-          Object[] arguments,
-          @CachedLibrary(limit = "10") TypesLibrary typesLibrary,
-          @Cached MethodResolverNode methodResolverNode) {
+      VirtualFrame frame,
+      State state,
+      UnresolvedSymbol symbol,
+      Object self,
+      Object[] arguments,
+      @CachedLibrary(limit = "10") TypesLibrary typesLibrary,
+      @Cached MethodResolverNode methodResolverNode) {
     Type selfTpe = typesLibrary.getType(self);
     Function function = resolveFunction(symbol, selfTpe, methodResolverNode);
     if (function == null) {
       throw new PanicException(
-              EnsoContext.get(this).getBuiltins().error().makeNoSuchMethod(self, symbol), this);
+          EnsoContext.get(this).getBuiltins().error().makeNoSuchMethod(self, symbol), this);
+    }
+    var resolvedFuncArgCount = function.getSchema().getArgumentsCount();
+    CallArgumentInfo[] invokeFuncSchema = invokeFunctionNode.getSchema();
+    // Check if this is a static method call on Any
+    if (isAnyEigenType(selfTpe) && arguments.length == resolvedFuncArgCount - 1) {
```

Comment:
I think this is correct. We want to find out if a method is defined on `Any` and treat it in a special way.